### PR TITLE
Add minMappingQuality option, improved coverage mouseover

### PR DESCRIPTION
**Pull request by alexander-veit**
- Moved the chromInfo functions to `chrominfo-utils.js`
- Added the option `minMappingQuality`. If this is set, reads with a mapping quality lower than the specified value are not displayed.
- When hovering over the coverage, the genomic range / position that is covered by the bar is now displayed

### CODE REVIEW

File: src/bam-fetcher-worker.js [hunk 1]
```diff
@@ -404,6 +298,23 @@ const getCoverage = (segmentList, samplingDistance) => {
     });
   }
 
+  // Add thousands separators for easier reading
+  const formatNumber = (num) => String(num).replace(/(.)(?=(\d{3})+$)/g,'$1,');
```
Comment (pkerpedjiev) [inline src/bam-fetcher-worker.js:302]: Might be better to use something like d3-format which offers more flexibility and legibility.

File: src/bam-fetcher-worker.js [hunk 2]
```diff
@@ -952,7 +867,7 @@ const renderSegments = (
       Math.floor((maxPos - minPos) / maxCoverageSamples),
       1,
     );
-    const result = getCoverage(segmentList, coverageSamplingDistance);
+    const result = getCoverage(uid, segmentList, coverageSamplingDistance);
```
Comment (pkerpedjiev) [inline src/bam-fetcher-worker.js:870]: What is `uid` here?

File: src/bam-fetcher-worker.js [hunk 3]
```diff
@@ -360,7 +253,7 @@ const serverTilesetInfo = (uid) => {
     });
 };
 
-const getCoverage = (segmentList, samplingDistance) => {
+const getCoverage = (uid, segmentList, samplingDistance) => {
```
Comment (pkerpedjiev) [inline src/bam-fetcher-worker.js:256]: Is `uid` actually used for anything?

Comment (pkerpedjiev) [reply src/bam-fetcher-worker.js:256]: Oh, I see, it's used further below.

File: src/bam-fetcher-worker.js [hunk 4]
```diff
@@ -191,7 +84,7 @@ const bamRecordToJson = (bamRecord, chrName, chrOffset) => {
     chrName,
     chrOffset,
     cigar: bamRecord.get('cigar'),
-    mapq: bamRecord.get('mq'),
+    mq: bamRecord.get('mq'),
```
[the remark this answers is not in the document]
Comment (alexander-veit) [reply src/bam-fetcher-worker.js:87]: I thought it is better to use a consistent naming (as in the bamRecord), but I don't have a strong preference.

File: src/bam-fetcher-worker.js [hunk 5]
```diff
@@ -382,6 +275,7 @@ const getCoverage = (segmentList, samplingDistance) => {
             T: 0,
             N: 0,
           },
+          range: "" // Will be used in the mouseover of the coverage track
```
[the remark this answers is not in the document]
Comment (alexander-veit) [reply src/bam-fetcher-worker.js:278]: See screenshot below

File: src/bam-fetcher-worker.js (hunk 1 above)
Comment (alexander-veit) [reply src/bam-fetcher-worker.js:302]: Agreed

File: src/bam-fetcher-worker.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (pkerpedjiev) [reply src/bam-fetcher-worker.js:87]: If you don't mind, let's leave it the way it is (`mapq`). The server component uses `mapq` and it seems like a more logical name:

https://github.com/higlass/clodius/blob/develop/clodius/tiles/bam.py#L115

File: src/bam-fetcher-worker.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (pkerpedjiev) [reply src/bam-fetcher-worker.js:278]: Would you mind adding something to the effect of "Will be used to show the bounds of this coverage bin when mousing over", just to make it clear what it's actually doing.

File: src/bam-fetcher-worker.js (hunk 2 above)
Comment (pkerpedjiev) [reply src/bam-fetcher-worker.js:870]: Ah yes. Makes sense 👍  A comment would be helpful. You can see how poor my memory is and I wrote a bunch of the pileup code :-)

**Pull request merged**